### ISSUE TITLE
elementorFrontend.utils is only working as expected when wrapped into setTimeout even with 0ms timeout

### ISSUE THREAD

**Issue by rloes** [elementor/elementor#21448]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When creating a custom Swiper on my page using the elementorFrontend.utils.swiper class following the guide [here](https://developers.elementor.com/experiment-optimized-asset-loading/), i am getting the error ```elementorFrontend.utils is not defined```. However, if I wrap the function described in the article under "Backward Compatibility For External Plugins Developers", in a setTimeout function with a Timeout of 0 seconds it magically works.

### Steps to reproduce

1) create custom swiper using follwing code
```javascript 
jQuery(window).on('elementor/frontend/init',function() {
		const swiperElement = document.getElementById('category-swiper_init');
		const swiperConfig = {
			slidesPerView: 1,
  		spaceBetween: 50,
			pagination: {
				el: '.category-swiper-pagination',
				clickable: true,
				renderBullet: function (index, className) {
          return '<span class="'+className+'">' + (index + 1) + "</span>";
        }
        },
			loop: true,
			speed: 1200,
			autoplay: {
				delay: 7000,
				disableOnInteraction: false
			}
		};
		if ( 'undefined' === typeof Swiper ) {
        const asyncSwiper = elementorFrontend.utils.swiper;

        new asyncSwiper( swiperElement, swiperConfig ).then( ( newSwiperInstance ) => {
            console.log( 'New Swiper instance is ready: ', newSwiperInstance );

            mySwiper = newSwiperInstance;
        });
    } else {
        console.log( 'Swiper global variable is ready, create a new instance: ', Swiper );

        mySwiper = new Swiper( swiperElement, swiperConfig );
    }
	})
```
This will produce the error described

2) If I wrap the initialization in a setTimeout with 0 seconds like so:
```javascript
jQuery(window).on('elementor/frontend/init',function() {
		const swiperElement = document.getElementById('category-swiper_init');
		const swiperConfig = {
			slidesPerView: 1,
  		spaceBetween: 50,
			pagination: {
				el: '.category-swiper-pagination',
				clickable: true,
				renderBullet: function (index, className) {
          return '<span class="'+className+'">' + (index + 1) + "</span>";
        }
        },
			loop: true,
			speed: 1200,
			autoplay: {
				delay: 7000,
				disableOnInteraction: false
			}
		};
		setTimeout(() =>{
						if ( 'undefined' === typeof Swiper ) {
				        const asyncSwiper = elementorFrontend.utils.swiper;
				
				        new asyncSwiper( swiperElement, swiperConfig ).then( ( newSwiperInstance ) => {
				            console.log( 'New Swiper instance is ready: ', newSwiperInstance );
				
				            mySwiper = newSwiperInstance;
				        });
				    } else {
				        console.log( 'Swiper global variable is ready, create a new instance: ', Swiper );
				
				        mySwiper = new Swiper( swiperElement, swiperConfig );
				    }}, 0)
	})
```
Now it works, but I do not understand why and I think this is not expected behaviour.

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Debian 11 v10.5.18
	PHP Version: 8.0.25
	PHP Memory Limit: 256M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 150M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.1.1
	Site URL: http://basisdruck.westsite-webdesign.de
	Home URL: http://basisdruck.westsite-webdesign.de
	WP Multisite: No
	Max Upload Size: 150 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: de-DE
	Timezone: Europe/Berlin
	Admin Email: intern@westsite-webdesign.de
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1.1676691042
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.6.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: de_DE
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/110.0

== Active Plugins ==
	Advanced Custom Fields
		Version: 6.0.7
		Author: WP Engine

	Better Elementor Excerpt
		Version: 1.0.0
		Author: Patryk Kożuch

	Child Theme Configurator
		Version: 2.6.0
		Author: Lilaea Media

	Code Snippets
		Version: 3.2.2
		Author: Code Snippets Pro

	Copy & Delete Posts
		Version: 1.3.8
		Author: Copy Delete Posts

	Custom Post Type UI
		Version: 1.13.4
		Author: WebDevStudios

	Elementor
		Version: 3.11.2
		Author: Elementor.com

	Elementor Pro
		Version: 3.11.2
		Author: Elementor.com

	Germanized for WooCommerce
		Version: 3.11.3
		Author: vendidero

	Newsletter
		Version: 7.6.6
		Author: Stefano Lissa & The Newsletter Team

	One Stop Shop for WooCommerce
		Version: 1.3.4
		Author: vendidero

	PDF Invoices & Packing Slips for WooCommerce
		Version: 3.4.0
		Author: WP Overnight

	The SEO Framework
		Version: 4.2.8
		Author: The SEO Framework Team

	WooCommerce
		Version: 7.4.0
		Author: Automattic

	WooCommerce Shipping & Tax
		Version: 2.2.3
		Author: WooCommerce

== Eigenschaften ==
	Custom Fonts: 1
	Custom Icons: 0

== Integrations ==
	
	woocommerce: Aktiv

== Elementor-Experimente ==
	Optimierte DOM Ausgabe: Standardmäßig aktiviert
	Verbessertes Laden von Assets: Standardmäßig aktiviert
	Verbessertes Laden von CSS: Standardmäßig aktiviert
	Inline-Schriftarten-Symbole: Aktiv
	Verbesserungen der Zugänglichkeit: Standardmäßig aktiviert
	Zusätzliche benutzerdefinierte Breakpoints: Standardmäßig aktiviert
	admin_menu_rearrangement: Standardmäßig deaktiviert
	Flexbox Container: Aktiv
	Upgrade Swiper Library: Aktiv
	Default to New Theme Builder: Standardmäßig aktiviert
	Hello Theme Header & Footer: Standardmäßig aktiviert
	Editor V2: Standardmäßig deaktiviert
	Startseiten: Standardmäßig aktiviert
	Nested Elements: Aktiv
	Lazy Load Background Images: Aktiv
	Als Standard speichern: Standardmäßig aktiviert
	Page Transitions: Standardmäßig aktiviert
	Notes: Standardmäßig aktiviert
	Loop: Aktiv
	Form Submissions: Standardmäßig aktiviert
	Scroll Snap: Standardmäßig aktiviert
	Mega Menu: Standardmäßig deaktiviert

== Protokoll ==
	
Log: showing 20 of 282023-02-16 14:11:39 [info] elementor::elementor_updater Started 
2023-02-16 14:11:39 [info] Elementor/Upgrades - _on_each_version Start  
2023-02-16 14:11:39 [info] Elementor/Upgrades - _on_each_version Finished 
2023-02-16 14:11:39 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]
2023-02-16 14:11:39 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]
2023-02-16 14:11:41 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]
2023-02-17 00:45:47 [info] elementor-pro::elementor_pro_updater Started 
2023-02-17 00:45:47 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-02-17 00:45:47 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-02-17 00:45:47 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]
2023-02-17 00:45:47 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]
2023-02-23 03:36:20 [info] elementor-pro::elementor_pro_updater Started 
2023-02-23 03:36:20 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-02-23 03:36:20 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-02-23 03:36:20 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.1',
  'to' => '3.11.2',
)]
2023-02-23 03:36:20 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.1',
  'to' => '3.11.2',
)]
2023-02-23 13:50:23 [info] elementor::elementor_updater Started 
2023-02-23 13:50:23 [info] Elementor/Upgrades - _on_each_version Start  
2023-02-23 13:50:23 [info] Elementor/Upgrades - _on_each_version Finished 
2023-02-23 13:50:23 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.1',
  'to' => '3.11.2',
)]

PHP: showing 6 of 6PHP: 2023-02-16 14:20:18 [warning X 3][/home/www/basisdruck/wp-content/plugins/elementor/includes/managers/image.php::108] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-02-16 20:22:10 [warning X 7][/home/www/basisdruck/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::168] Undefined array key "editor_post_id" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-02-17 00:35:09 [warning X 33][/home/www/basisdruck/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key "condition_type" [array (
  'trace' => '
#0: /home/www/basisdruck/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: /home/www/basisdruck/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: /home/www/basisdruck/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: /home/www/basisdruck/wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2023-02-18 16:01:22 [warning X 14][/home/www/basisdruck/wp-content/plugins/elementor/core/common/modules/ajax/module.php::175] Undefined array key "data" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-02-19 03:12:05 [error X 5][/home/www/basisdruck/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::295] Uncaught Error: Call to a member function update_meta() on bool in /home/www/basisdruck/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:295
Stack trace:
#0 /home/www/basisdruck/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(168): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(NULL, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /home/www/basisdruck/wp-content/plugins/elementor/core/common/modules/ajax/module.php(175): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 /home/www/basisdruck/wp-includes/class-wp-hook.php(308): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 /home/www/basisdruck/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#5 /home/www/basisdruck/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#6 /home/www/basisdruck/wp-admin/admin-ajax.php(188): do_action('wp_ajax_element...')
#7 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-02-19 21:22:28 [warning X 68][/home/www/basisdruck/wp-content/plugins/elementor-pro/modules/woocommerce/widgets/cart.php::2383] Undefined array key "section_coupon_display" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 8 of 8JS: 2023-02-16 19:01:45 [error X 2][http://basisdruck.westsite-webdesign.de/wp-content/plugins/elementor-pro/assets/js/editor.min.js?ver=3.11.0:2:61305] this.getEditorControlView(...) is undefined 
JS: 2023-02-16 19:16:53 [error X 2][http://basisdruck.westsite-webdesign.de/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.1:3:791125] T.originalEvent.key is undefined 
JS: 2023-02-17 01:35:59 [error X 3][http://basisdruck.westsite-webdesign.de/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.1:3:791626] this.store.find(...) is undefined 
JS: 2023-02-17 02:27:38 [error X 6][http://basisdruck.westsite-webdesign.de/:504:23] elementorFrontend.utils is undefined 
JS: 2023-02-18 15:01:11 [error X 2][http://basisdruck.westsite-webdesign.de/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.1:3:607993] this.container is undefined 
JS: 2023-02-19 13:27:23 [error X 2][http://basisdruck.westsite-webdesign.de/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.1:3:676178] C.getControlModel(...) is undefined 
JS: 2023-02-20 21:47:21 [error X 2][http://basisdruck.westsite-webdesign.de/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.1:3:676196] Cannot read properties of undefined (reading 'get') 
JS: 2023-02-23 13:11:56 [error X 2][http://basisdruck.westsite-webdesign.de/wp-content/plugins/elementor/assets/js/frontend.min.js?ver=3.11.2:2:29508] e.closest is not a function 

== Elementor - Compatibility Tag ==
	
	Better Elementor Excerpt: Die Kompatibilität ist nicht angegeben
	Elementor Pro: Die Kompatibilität ist nicht angegeben

== Elementor Pro - Compatibility Tag ==

**Comment by nicholaszein** [comment]
Hello! 

Thank you for submitting your issue! 🙏 

Please allow us some time to investigate it, and get back to you with our findings. 🧐

Best regards

**Comment by latheeshvm** [comment]
Facing the same issue. Am getting this error `e.closest is not a function` without setTimeout

**Comment by rloes** [comment]
@latheeshvm I got the `e.closest` error, when passing a css selector as swiperElement to ```new asyncSwiper( swiperElement, swiperConfig )``` instead of a reference to a DOM node. Maybe this works for you

**Comment by latheeshvm** [comment]
@rloes Thanks, I had a recheck. Now it's working without setTimeOut. The error shows up because of the HTML structure. 

check if you have the html structure with classes like below
```
<!-- Slider main container -->
<div id="category-swiper_init">
  <!-- Additional required wrapper -->
  <div class="swiper-wrapper">
    <!-- Slides -->
    <div class="swiper-slide">Slide 1</div>
    <div class="swiper-slide">Slide 2</div>
    <div class="swiper-slide">Slide 3</div>
    ...
  </div>
  <!-- If we need pagination -->
  <div class="swiper-pagination"></div>

  <!-- If we need navigation buttons -->
  <div class="swiper-button-prev"></div>
  <div class="swiper-button-next"></div>

  <!-- If we need scrollbar -->
  <div class="swiper-scrollbar"></div>
</div>
```
